### PR TITLE
docs: add common terms documentation to website

### PR DESCRIPTION
**Pull request by alwaysintreble**
@Ijwu @LegendaryLinux some website documentation. Let me know if there's anything you believe is missing.

### CODE REVIEW

**Comment by SoldierofOrder** [comment]
I have some suggestions for a few of the terms:

- I think we should reorder the first three terms to go Item -> Location -> Check instead of Check -> Item -> Location. The Check definition references locations, and the Location definition references Items, so a brand newbie might have an easier time parsing things if they read the Items definition first.

- I think for the Items definition, we should avoid mentioning the server (I totally recognize if y'all have a reason why we should mention it! I'm just under the impression that "item" can be defined without reference to online or multiplayer functionality). Here's my own attempt to rephrase: "Items are the things that get shuffled around (in other words, "randomized") as part of using a randomizer."
- For the Items definition, if we still want to mention something about how items work in Archipelago specifically, maybe we add something like "In Archipelago, your items aren't just randomized within your own game. If you've set up an Archipelago multiworld, all items are shuffled between all of the worlds."

- Maybe I'm off base here, but I feel like "check" as a _verb_ is very intuitive for native English speakers, e.g., "have you checked the basin fountain" makes sense for anyone who knows what "the basin fountain" means. I think the weirdness (and thus, the need to define it) comes from when people use it as a noun, e.g. "i gotta go back for the six checks I missed in Deepnest". I feel like this definition, as written, is trying to define it as a verb, so I think it's worth revising. I hadn't heard this word until I joined this community, like, a few days ago, so I'll leave my thoughts out on this one unless prompted, since I'm sure y'all have a better sense of what it means than I do.

I have a much weaker grasp on the other terms, so I'll come back to those if you want me to. :)

File: WebHostLib/static/assets/faq/common-terms_en.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+Items are what you will receive from the server regardless of what game you are playing. This could be a sword, a stat 
+upgrade, a spell, or any other potential receivable for your game.
+
+## Location
+Locations are where items are placed in your game. Whenever you interact with a location, you or another player will
+then receive an item. A location could be a chest, an enemy drop, a shop purchase, or any other interactable that can
+contain items in your game.
+
+## Slot
+A slot is the player name and number assigned during generation. This is equal to the number of `yaml` setting files
```
Comment (ThePhar) [inline WebHostLib/static/assets/faq/common-terms_en.md:21]: This is not entirely accurate, just because `yaml` setting files can have multiple yamls now. Maybe a small tweak to remove reference to `files`?

File: WebHostLib/static/assets/faq/common-terms_en.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+## Room
+Whenever you generate a seed on the Archipelago website you will be put on a seed page that contains all the seed info
+with a link to the spoiler if one exists and will show how many unique rooms exist per seed. Each room has its own 
+unique identifier that is separate from the seed. The room page is where you can find information to connect to the
+multiworld and download any patches if necessary. If you have a particularly fun or interesting seed, and you want to 
+share it with somebody you can link them to this seed page, where they can generate a new room to play it! For seeds 
+generated with race mode enabled, the seed page will only show rooms created by the unique user so the seed page is perfectly safe to share for racing purposes.
+
+## Logic
```
Comment (ThePhar) [inline WebHostLib/static/assets/faq/common-terms_en.md:52]: I'd probably add a note that something can be considered "out of logic" if accessing a location requires the use of special knowledge that the randomizer does not take into account when calculating if you can access the location.

**Review by ThePhar — COMMENTED** [review]
Couple key things that might be nice to tweak, in my humble opinion.

File: WebHostLib/static/assets/faq/common-terms_en.md (hunk 1 above)
Comment (alwaysintreble) [reply WebHostLib/static/assets/faq/common-terms_en.md:21]: I think the original intent here I was planning on covering the case of using a weightedsettings.yaml but decided it was out of scope. will reevaluate this though

File: WebHostLib/static/assets/faq/common-terms_en.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## Progression
+Certain items will allow access to more locations and are considered progression items as they "progress" the seed.
+
+## Trash
+A term used for "filler" items that have no bearing on the generation and are either marginally useful for the player
+or useless. These items can be very useful depending on the player but are never very important and as such are usually
+termed trash.
+
+## Burger King / BK Mode
```
Comment (LegendaryLinux) [inline WebHostLib/static/assets/faq/common-terms_en.md:66]:
```md
## Burger King / BK Mode / BK
A term used to indicate when a player has no checks available to them, and is awaiting progression from another player. There are two types of BK Mode:  
1. "Logical BK" refers to when a player has no logical checks remaining, but does have the ability to acquire some checks out of logic.  
2. "Hard BK" refers to when a player has no checks remaining in or out of logic.

The origin of this term comes from a GDQ (Games Done Quick) event in which many players were participating in an Ocarina of Time MultiWorld. One player found themselves in hard BK mode, and decided to visit the Burger King attached to the hotel. That player returned and finished their meal before receiving a progression item.
```

File: WebHostLib/static/assets/faq/common-terms_en.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# Common Multiworld Terms
+
+There are a lot of common terms used when playing in different game randomizer communities and in multiworld as well.
+This document serves as a lookup for common terms that may be used by users in the community or in various other
```
Comment (LegendaryLinux) [inline WebHostLib/static/assets/faq/common-terms_en.md:4]: Line 4 is unnecessary. The title of the document indicates its purpose.

File: WebHostLib/static/assets/commonTerms.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+window.addEventListener('load', () => {
+    const tutorialWrapper = document.getElementById('common-terms-wrapper');
+    new Promise((resolve, reject) => {
+        const ajax = new XMLHttpRequest();
+        ajax.onreadystatechange = () => {
+            if (ajax.readyState !== 4) { return; }
+            if (ajax.status === 404) {
+                reject("Sorry, the tutorial is not available in that language yet.");
```
Comment (LegendaryLinux) [inline WebHostLib/static/assets/commonTerms.js:8]: "Sorry, the jargon file is not available in that language yet."

File: WebHostLib/static/assets/commonTerms.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                return;
+            }
+            if (ajax.status !== 200) {
+                reject("Something went wrong while loading the tutorial.");
```
Comment (LegendaryLinux) [inline WebHostLib/static/assets/commonTerms.js:12]: "Something went wrong while loading the jargon file."

File: WebHostLib/__init__.py [hunk 7]
```diff
@@ -141,6 +141,11 @@ def faq(lang):
     return render_template("faq.html", lang=lang)
 
 
+@app.route('/common-terms/<string:lang>/')
```
Comment (LegendaryLinux) [inline WebHostLib/__init__.py:144]:
```py
@app.route('jargon/<string:lang>/')
```

**Review by LegendaryLinux — CHANGES_REQUESTED** [review]
I left a bunch of comments indicating requested changes to language. Overall this is a pretty good job. Thanks for writing this up!

File: WebHostLib/templates/siteMap.html [hunk 8]
```diff
@@ -26,6 +26,7 @@ <h2>Base Pages</h2>
             <li><a href="/user-content">User Content</a></li>
             <li><a href="/weighted-settings">Weighted Settings Page</a></li>
             <li><a href="{{url_for('stats')}}">Game Statistics</a></li>
+            <li><a href="/common-terms">Common Multiworld Terms</a></li>
```
[the remark this answers is not in the document]
Comment (alwaysintreble) [reply WebHostLib/templates/siteMap.html:29]: I almost used the word Jargon which would be more "correct" but I figured it might confuse users who don't know what Jargon means so opted to use "common terms" as not necessarily everything documented here is jargon

**Comment by alwaysintreble** [comment]
i've made some changes and updates. would like to get @SoldierofOrder and @mDuo13 opinions on this

File: WebHostLib/static/assets/faq/common-terms_en.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+contain items in your game.
+
+## Check
+A check is a common term for when you "check", or pick up, a location. In terms of Archipelago this is usually used for
+when a player goes to a location and sends its item, or "checks" the location. Players will often reference their now
+randomized locations as checks.
+
+## Slot
+A slot is the player name and number assigned during generation. The number of slots is equal to the number of players,
+or "worlds", created.
```
Comment (mDuo13) [inline WebHostLib/static/assets/faq/common-terms_en.md:23]: Maybe note that each player needs a unique name?

**Comment by LegendaryLinux** [comment]
I will not approve this until you use the word `jargon` as the name of your jargon file.

Your arguments against the word are:
1. Non-English speakers might not understand the word
2. Some English speakers might not understand the word

My responses are:
1. The page being loaded is the English version, as described in your route
2. We shouldn't just decide not to use words because people don't know what they mean. Some people don't know what an Archipelago is, but that's our name. Teach people, don't just leave them in ignorance.

File: WebHostLib/static/assets/faq/common-terms_en.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+## World
+World in terms of Archipelago can mean multiple things and is used interchangeably in many situations.
+* During gameplay in multiworlds each world is a "slot" or "player." As Archipelago allows multiple players to connect
+to the same slot those players are now sharing a world and completing it cooperatively. For games with native 
+cooperative play Archipelago usually fully supports this and those players would connect to each other and only one
+of them to the multiworld but they would still be sharing the same world.
+* For the sake of the backend, or programming side, world is usually intended to represent the world package for a game 
+integration. For example this could be the entire `worlds/factorio` directory.
```
Comment (mDuo13) [inline WebHostLib/static/assets/faq/common-terms_en.md:32]: I think this definition can be tightened up a bit to clarify the important parts more concisely. Here's an attempt:

```suggestion
World in terms of Archipelago can mean multiple things and is used interchangeably in many situations.
* During gameplay, a world is a single instance of a game, occupying one player "slot". However, 
Archipelago allows multiple players to connect to the same slot; then those players can share a world 
and complete it cooperatively. For games with native cooperative play, you can also play together and
share a world that way, usually with only one player connected to the multiworld.
* On the programming side, a world typically represents the package that integrates Archipelago with a
particular game. For example this could be the entire `worlds/factorio` directory.
```

File: WebHostLib/static/assets/faq/common-terms_en.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+## RNG
+Acronym for "Random Number Generator." Archipelago uses its own custom Random object with a unique seed per generation,
+or, if running from source, a seed can be supplied and this seed will control all randomization during generation as all
+game worlds will have access to it.
+
+## Seed
+A "seed" is a number used to initialize a pseudorandom number generator. Whenever you generate a new game on Archipelago
+this is a new "seed" as it has unique item placement, and you can create multiple "rooms" on the Archipelago site from a
+single seed.
```
Comment (mDuo13) [inline WebHostLib/static/assets/faq/common-terms_en.md:42]:
```suggestion
A "seed" is a number used to initialize a pseudorandom number generator. Whenever you generate a new game on Archipelago
this is a new "seed" as it has unique item placement, and you can create multiple "rooms" on the Archipelago site from a
single seed. Using the same seed results in the random placement being the same.
```

**Review by mDuo13 — APPROVED** [review]
Definitely a helpful improvement! I left some suggestions, but feel free to consider them optional.

**Comment by LegendaryLinux** [comment]
> to non-native speakers

This is the English version of the jargon file. Its hyperlink literally ends with `/en`

As for definition, let's just reference dictionary.com for this:  
https://www.dictionary.com/browse/jargon
> the language, especially the vocabulary, peculiar to a particular trade, profession, or group

> language that is characterized by uncommon or pretentious vocabulary and convoluted syntax and is often vague in meaning

https://www.dictionary.com/browse/glossary
> a list of terms in a special subject, field, or area of usage, with accompanying definitions

> such a list at the back of a book, explaining or defining difficult or unusual words and expressions used in the text

While glossary has a similar definition, their uses are distinct. A glossary is typically found at the end of a textbook or technical writing, while jargon references a set of common terms used in a specific field or group.

Jargon is the correct word for this file.

**Comment by mDuo13** [comment]
> > to non-native speakers
> 
> This is the English version of the jargon file. Its hyperlink literally ends with `/en`

People whose native language is not available may read the English version if their primary language is not available, which in most cases it is not. Personally, I'd rather provide a welcoming experience to a broader audience than gatekeep people based on their language proficiency.

> While glossary has a similar definition, their uses are distinct. A glossary is typically found at the end of a textbook or technical writing, while jargon references a set of common terms used in a specific field or group.

"Jargon" describes the words in the file. "Glossary" describes the file. Both are appropriate, but "glossary" is a better name for the file itself _and_ it's more commonly used in documentation, which this is. Also, [jargon file](https://en.wikipedia.org/wiki/Jargon_File) is the name of a specific notable publication, so unless you're trying to pay tribute to that instance, I'd say the word "glossary" is more natural.

That's the last I'll say on this. This PR represents real improvements on the beginner documentation and I'd rather not hold those up with an extended argument about semantics.

**Comment by SoldierofOrder** [comment]
I agree with @mDuo13 on these changes. This is a great foundation (for which I'm, of course, still going to have notes + suggestions :) but that comes later, once we have these approved), and I don't think it's worth delaying for this.

I'll weigh in on the discussion, though, in hopes of persuading @LegendaryLinux. First, on a philosophical note: we don't have extensive support for the documentation in all common languages, as I understand it, and the dominant language of our community is English. Therefore, people _will_ come to us only knowing a language we don't support, defaulting to English instead, and it's worth choosing language that's more accessible to them even when it conflicts with language that would suit our (very individual!) aesthetic sensibilities.

Then, on a definitional note: the Wikipedia page mDuo13 links to about the "Jargon File" even specifies that that file **is** a glossary. Based on the definitions you posted, I believe that anything that can be called a "jargon file" can also be called a glossary (even if the reverse is not necessarily true). So I see little reason to prefer one over the other if we're going purely based off of dictionary definitions.

For what it's worth, I approve this documentation as it stands, with the intention to suggest further changes later. :)

**Review by LegendaryLinux — CHANGES_REQUESTED** [review]
It seems like I'm in the minority. `Glossary` is better than `Common Terms`, at least. I'll approve once the page title and route are changed to `Glossary` and `/glossary/en` respectively.

**Comment by alwaysintreble** [comment]
> For what it's worth, I approve this documentation as it stands, with the intention to suggest further changes later. :)

Since this is open source and this document is just an md you can always just PR more stuff in yourself if you feel inclined. I'm not a super great writer so I have no issue with people editing my text as long as I don't feel that it makes the understanding more complicated.

**Review by LegendaryLinux — APPROVED** [review]
